src/circular-queue.test.js

```diff
@@ -0,0 +1,100 @@
+var MyCircularQueue = require('./circular-queue');
+
+describe('MyCircularQueue', function() {
```

Comment:
I love this style of testing. Have you seen anything like this for Java? Or Go?

Reply:
What do you mean by "this style of testing"? Kind've `RSpec`-y?

Reply:
Yes. It's very descriptive and easy to understand what's going on.
Not sure if it's because the test and use cases are simple.
DHH started a big debate around it a long time ago. [This article explains a bit](http://www.rubyinside.com/dhh-offended-by-rspec-debate-4610.html).
There are many perspectives on behavior driven testing. Some days I feel like I want to use it, then others, I fell like just writing good unit-test-style tests. 🤷‍♂️